### PR TITLE
Add unit tests for ApiErrorAndDescriptionToStatus

### PR DESCRIPTION
**Pull request by dmondrik**
### What does this Pull Request accomplish?

Adds unit tests for `ApiErrorAndDescriptionToStatus` which was added in the previous PR.

Getting the previous PR to a sufficiently passing state required modifying tests to *indirectly* ensure its behavior, but this is a more direct verification.

### Why should this Pull Request be merged?

[Task 2103383](https://ni.visualstudio.com/DevCentral/_workitems/edit/2103383): Ensure description strings are properly JSON encoded.

### What testing has been done?

Built locally and ran the unit tests successfully.

### CODE REVIEW

**Review by reckenro — APPROVED** [review]

**Review by astarche — APPROVED** [review]

File: source/tests/unit/converters_tests.cpp [hunk 1]
```diff
@@ -170,6 +172,41 @@ TEST(ConvertersTests, ArrayAndRawValue_ConvertBitfieldAsEnumArrayInput_ReturnsOr
...
+  nlohmann::json jsonError{{"custom", 1}};
+
+  auto error_message = nidevice_grpc::ApiErrorAndDescriptionToStatus(0, empty_description, jsonError).error_message();
+
+  EXPECT_THAT(error_message, HasSubstr("\"custom\":1"));
+}
+
+TEST(ConvertersTests, QuotedLocalizedDescription_ApiErrorAndDescriptionToStatus_MessageIsEncoded)
+{
+  const auto TEST_MESSAGE = std::string("\"chaîne 😀 localisée\"");
```
Comment (bkeryan) [inline source/tests/unit/converters_tests.cpp:200]: What encoding are you using in this source file?

MSVC has command line options to specify that the source code is UTF-8: https://docs.microsoft.com/en-us/cpp/build/reference/utf-8-set-source-and-executable-character-sets-to-utf-8?view=msvc-170

Hex escapes are more portable: `"\cha\xC3\xAEe \xF0\x9F\x98\x80 localis\xC3\xA9e\""`

Comment (dmondrik) [reply source/tests/unit/converters_tests.cpp:200]: Readability is important too. I found that specifying the string prefix `u8` will guarantee it, so I'll add that to my next PR.

Comment (bkeryan) [reply source/tests/unit/converters_tests.cpp:200]: Based on https://docs.microsoft.com/en-us/cpp/build/reference/utf-8-set-source-and-executable-character-sets-to-utf-8?view=msvc-170 I would not be surprised if MSVC interprets the source as Windows-1252 and then converts the `u8` strings from Windows-1252 to UTF-8. :)

`u8` strings also support the `\uNNNN` escape.

Comment (dmondrik) [reply source/tests/unit/converters_tests.cpp:200]: So our C++ standard representative tells me passing a u8 literal to std::string will stop compiling as of C++20, so I guess I can't do that. Oh well.

Comment (bkeryan) [reply source/tests/unit/converters_tests.cpp:200]: Interesting: https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2019/p1423r0.html

> Code surveys have so far revealed little use of u8 literals. Google and Facebook have both reported less than 1000 occurrences in their code bases, approximately half of which occur in test code. Representatives of both organizations have stated that, given the actual size of their code base, this is approximately equivalent to 0.
>
> Searches on Debian code search found uses in only a few packages and, within those packages, a small number of uses (mostly single digit use counts), most of which occurred in tests.

**Pull request merged**